### PR TITLE
Bicycle integrity fixes

### PR DESCRIPTION
**Pull request by RandomGamer123**
## About The Pull Request
Code copied from ATV code, makes bicycles not immediately break upon taking damage and allow them to be fixed with a welder.

## Why It's Good For The Game
So, let's imagine you are a good hardworking plasmeme who just sold 13000 moles of nitrium and bought yourself a brand new 1000000 credit worth bicycle, you drive it around, make some bombs while riding it, kill a malf AI while riding it, and then all of a sudden, it breaks, because you slipped on some ice and collided with it, and you can't even fix it despite having a welder on you.
unfixablebike

On a more serious justification, bicycles not having welder_act seems like an oversight because of how it is out of line from how other player-obtainable vehicles (eg. atv / secway) behave, and it doesnt really make sense that a 1000000 credit bicycle breaks so easily and cant be fixed.

## Changelog

:cl:
fix: Bicycles can now be repaired with a welder and have actually sensible max_integrity and integrity_failure values.
/:cl:

### CODE REVIEW

**Comment by ArcaneDefence** [comment]
> it doesnt really make sense that a 1000000 credit bicycle breaks so easily and cant be fixed.

How else is Centcom supposed to sell you another?

File: code/modules/vehicles/bicycle.dm [hunk 1]
```diff
@@ -16,3 +18,14 @@
 	. = ..()
 	for(var/m in buckled_mobs)
 		unbuckle_mob(m,1)
+
+/obj/vehicle/ridden/bicycle/welder_act(mob/living/user, obj/item/I)
+	if(atom_integrity >= max_integrity)
+		return TRUE
+	if(!I.use_tool(src, user, 0, volume=50, amount=1))
+		return TRUE
+	user.visible_message(span_notice("[user] repairs some damage to [name]."), span_notice("You repair some damage to \the [src]."))
```
Comment (Mothblocks) [inline code/modules/vehicles/bicycle.dm:27]:
```suggestion
	user.visible_message(span_notice("[user] repairs some damage to [src]."), span_notice("You repair some damage to [src]."))
```

File: code/modules/vehicles/bicycle.dm [hunk 2]
```diff
@@ -16,3 +18,14 @@
...
+	atom_integrity += min(10, max_integrity-atom_integrity)
+	if(atom_integrity == max_integrity)
+		to_chat(user, span_notice("It looks to be fully repaired now."))
```
Comment (Mothblocks) [inline code/modules/vehicles/bicycle.dm:30]:
```suggestion
	atom_integrity += min(10, max_integrity-atom_integrity)
	if(atom_integrity == max_integrity)
		balloon_alert(user, "fully repaired")
	else
		balloon_alert(user, "repaired some damages")
```

**Comment by Iamgoofball** [comment]
please make sure it's unfixable if it gets fried by electricity btw

**Comment by RandomGamer123** [comment]
> please make sure it's unfixable if it gets fried by electricity btw

Tested that it cant be and made explicitly clear with a special alert

File: code/modules/vehicles/bicycle.dm [hunk 3]
```diff
@@ -13,6 +16,22 @@
...
 	color = rgb(63, 23, 4)
 	can_buckle = FALSE
+	fried = TRUE
 	. = ..()
 	for(var/m in buckled_mobs)
 		unbuckle_mob(m,1)
+
+/obj/vehicle/ridden/bicycle/welder_act(mob/living/user, obj/item/I)
+	if(fried)
+		balloon_alert(user, "cannot repair a fried bicycle")
```
Comment (Mothblocks) [inline code/modules/vehicles/bicycle.dm:26]:
```suggestion
		balloon_alert(user, "it's fried!")
```

Short and to the point, `!` to indicate failure.

**Pull request merged**